app/services/email_service.rb

```diff
@@ -0,0 +1,86 @@
+class EmailService
+
+  attr_reader :client, :users, :type, :link
+
+  def initialize(client:, users:, type:, link: nil)
+    @client = client
+    @users = Array(users)
+    @type = type
+    @link = link
+  end
+
+  def send_email!
+    begin
+      response = client.send_email(template)
+      response.to_h
+    rescue Aws::SES::Errors::ServiceError => error
+      puts "Email not sent. Error message: #{error}"
+    end
+  end
+
+  private
+
+  def recipients
+    users.map { |u| u.email }
+  end
+
+  def subject
+    email_contents[:subject]
+  end
+
+  def html_body
+    return html_body_with_link if link
+    email_contents[:html_body]
+  end
+
+  # def personalized_body(first_name)
+  #   personalized_body = email_contents[:html_body].dup
+  #   personalized_body['<first_name>'] = first_name
+  # end
```

Comment:
Going to be needing this soon or can we remove?

Reply:
I'll be implementing this soon